src/jdk.javadoc/share/classes/jdk/internal/org/commonmark/internal/Bracket.java

```diff
@@ -0,0 +1,64 @@
+package jdk.internal.org.commonmark.internal;
+
+import jdk.internal.org.commonmark.internal.inline.Position;
+import jdk.internal.org.commonmark.node.Text;
+
+/**
+ * Opening bracket for links (<code>[</code>) or images (<code>![</code>).
+ */
+public class Bracket {
+
+    public final Text node;
+
+    /**
+     * The position of the marker for the bracket (<code>[</code> or <code>![</code>)
+     */
+    public final Position markerPosition;
+
+    /**
+     * The position of the content (after the opening bracket)
+     */
+    public final Position contentPosition;
+
+    /**
+     * Whether this is an image or link.
+     */
+    public final boolean image;
+
+    /**
+     * Previous bracket.
+     */
+    public final Bracket previous;
+
+    /**
+     * Previous delimiter (emphasis, etc) before this bracket.
+     */
+    public final Delimiter previousDelimiter;
+
+    /**
+     * Whether this bracket is allowed to form a link/image (also known as "active").
+     */
+    public boolean allowed = true;
+
+    /**
+     * Whether there is an unescaped bracket (opening or closing) anywhere after this opening bracket.
+     */
+    public boolean bracketAfter = false;
+
+    static public Bracket link(Text node, Position markerPosition, Position contentPosition, Bracket previous, Delimiter previousDelimiter) {
```

Comment:
Let's use blessed modifiers order
```suggestion
    public static Bracket link(Text node, Position markerPosition, Position contentPosition, Bracket previous, Delimiter previousDelimiter) {
```

Reply:
This class belongs to a 3rd party library, CommonMark. Changes that aren't necessary for the library to work in the JDK or not related to security might not be worth it.

Reply:
Agreed that we should not do "cosmetic" changes on imported code.